simpleclient_httpserver/src/test/java/io/prometheus/client/exporter/TestHTTPServer.java

```diff
@@ -115,316 +93,412 @@ public void init() throws IOException {
+                      ^ (httpResponse.getHeader("transfer-encoding") == null));
+
+      assertThat(body).contains("a 0.0");
+      assertThat(body).contains("b 0.0");
+      assertThat(body).doesNotContain("c 0.0");
     } finally {
-      s.close();
+      httpServer.close();
     }
   }
 
   @Test
-  public void testMultiName() throws IOException {
-    HTTPServer s = new HTTPServer(new InetSocketAddress(0), registry);
+  public void testSampleNameFilter() throws IOException {
+    HTTPServer httpServer = new HTTPServer.Builder()
+            .withRegistry(registry)
+            .withSampleNameFilter(new SampleNameFilter.Builder()
+                    .nameMustNotStartWith("a")
+                    .build())
+            .build();
+
+    HttpRequest httpRequest = new HttpRequest.Builder()
+            .withURL("http://localhost:" + httpServer.getPort() + "/metrics?name[]=a&name[]=b")
+            .build();
+
     try {
-      String response = request(s, "?name[]=a&name[]=b").body;
-      assertThat(response).contains("a 0.0");
-      assertThat(response).contains("b 0.0");
-      assertThat(response).doesNotContain("c 0.0");
+      HttpResponse httpResponse = httpRequest.execute();
+      String body = httpRequest.execute().getBody();
+
+      Assert.assertTrue(
+              (httpResponse.getHeader("content-length") == null)
+                      ^ (httpResponse.getHeader("transfer-encoding") == null));
+
+      assertThat(body).doesNotContain("a 0.0");
+      assertThat(body).contains("b 0.0");
+      assertThat(body).doesNotContain("c 0.0");
     } finally {
-      s.close();
+      httpServer.close();
     }
   }
 
   @Test
-  public void testSampleNameFilter() throws IOException {
-    HTTPServer s = new HTTPServer.Builder()
+  public void testSampleNameFilterEmptyBody() throws IOException {
+    HTTPServer httpServer = new HTTPServer.Builder()
             .withRegistry(registry)
             .withSampleNameFilter(new SampleNameFilter.Builder()
                     .nameMustNotStartWith("a")
+                    .nameMustNotStartWith("b")
                     .build())
             .build();
+
+    HttpRequest httpRequest = new HttpRequest.Builder()
+            .withURL("http://localhost:" + httpServer.getPort() + "/metrics?name[]=a&name[]=b")
+            .build();
+
     try {
-      String response = request(s, "?name[]=a&name[]=b").body;
-      assertThat(response).doesNotContain("a 0.0");
-      assertThat(response).contains("b 0.0");
-      assertThat(response).doesNotContain("c 0.0");
+      HttpResponse httpResponse = httpRequest.execute();
+
+      Assert.assertTrue(
+              (httpResponse.getHeader("content-length") == null)
+                      ^ (httpResponse.getHeader("transfer-encoding") == null));
+
+      String body = httpResponse.getBody();
+      assertThat(body).doesNotContain("a 0.0");
+      assertThat(body).doesNotContain("b 0.0");
+      assertThat(body).doesNotContain("c 0.0");
```

Comment:
Can be replaced with `assertThat(body).isEmpty()`.

Reply:
Agreed.